src/fciqmc_col.jl

```diff
@@ -30,25 +40,41 @@ fciqmc_col!(w::Union{AbstractArray,AbstractDVec}, args...) = fciqmc_col!(Stochas
 fciqmc_col!(::Type{T}, args...) where T = throw(TypeError(:fciqmc_col!,
     "first argument: trait not recognised",StochasticStyle,T))
 
+###
+### Concrete implementations
+###
+
+function step_stats(::IsDeterministic)
+    return (:exact_steps,), SVector(0,)
+end
 function fciqmc_col!(::IsDeterministic, w, ham::AbstractMatrix, add, num, shift, dτ)
-    w[add] += (1 + dτ * shift) * num # diagonal without Hamiltonian contribution
-    w .+= -dτ .* ham[:, add] .* num # full matrix column
-    # todo: return something sensible
-    return (0, 0, 0, 0, 0)
+    # diagonal without Hamiltonian contribution
+    for i in axes(ham, 1)
+        i == add && continue
+        deposit!(w, i, -dτ * ham[i, add] * num, add => num)
+    end
+    deposit!(w, add, (1 + dτ * (shift - ham[add, add])) * num, add => num)
+    return (1,)
```

Comment:
```suggestion
    for i in axes(ham, 1) # iterate through off-diagonal rows of `ham`
        i == add && continue
        deposit!(w, i, -dτ * ham[i, add] * num, add => num)
    end
    deposit!(w, add, (1 + dτ * (shift - ham[add, add])) * num, add => num) # diagonal
    return (1,)
```

Reply:
Looks cool! So does that mean we can run initiator fciqmc on an abstract matrix?